inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/metadata/MetadataFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.inlong.agent.plugin.metadata;
+
+import org.apache.inlong.agent.plugin.SourceMeta;
+
+import static org.apache.inlong.agent.constant.JobConstants.META_AGENT;
+import static org.apache.inlong.agent.constant.JobConstants.META_KUBERNETES;
+
+/**
+ * metadata factory
+ */
+public class MetadataFactory {
+
+    public static SourceMeta getInstance(String metaType) {
+        switch (metaType) {
+            case META_KUBERNETES:
+                return new KubernetesMetadata();
+            case META_AGENT:
+                return new AgentMetadata();
+            default:
+                return null;
```

Comment:
it's better to add a default value to avoid exceptions.

Reply:
To solve another PR